### PR TITLE
Add check to make sure folks are using npm 6.13 with playground

### PR DESCRIPTION
**Pull request by chrisfromwork**
This review contains the following changes:

1. The readme now contains a known issues section
2. running npm install in Apps\Playground now assesses the npm version folks are using to make sure they aren't trying to build with npm 7+

### CODE REVIEW

File: Package/gulpfile.js [hunk 1]
```diff
@@ -439,6 +439,18 @@ const createPackageUWP = async () => {
   exec('npm pack', 'Assembled-Windows');
 }
 
+const checkNPMVersion = async () => {
+  const output = shelljs.exec('npm list -g npm');
+  const parts = output.split("npm@");
+  const version = parts[1].trim();
+  
+  // Check if using npm version 7+
+  if (version.charAt(0) == '7')
```
Comment (chrisfromwork) [inline Package/gulpfile.js:448]: not sure if folks would prefer something more intelligent here.

File: Apps/Playground/scripts/postinstall.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+const shelljs = require('shelljs');
+const output = shelljs.exec('npm list -g npm');
+const parts = output.split("npm@");
+const version = parts[1].trim();
+
+// Check if using npm version 7+
+if (version.charAt(0) == '7')
```
Comment (rgerd) [inline Apps/Playground/scripts/postinstall.js:7]: You could also do something like `!!version.match(/6\.13\..*/g)`, or import semver

**Review by rgerd — APPROVED** [review]

File: Apps/Playground/scripts/postinstall.js (hunk 2 above)
Comment (chrisfromwork) [reply Apps/Playground/scripts/postinstall.js]: this should now be updated

File: Apps/Playground/scripts/postinstall.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+let output = shelljs.exec('npm list -g npm');
+
+if (!output.includes("npm@")) {
+    shelljs.echo("global npm install not found");
+    output = shelljs.exec('npm list npm');
+}
+
+if (!output.includes("npm@")) {
+    shelljs.echo("local npm version not found");
+
```
Comment (rgerd) [inline Apps/Playground/scripts/postinstall.js:11]: nit: line

**Comment by rgerd** [comment]
You could also in theory do something like:
```javascript
"preinstall": "npm --version | scripts/preinstall.js"
```

```javascript
function getStandardInput() {
    return new Promise((resolve, reject) => {
        const stdin = process.stdin;
        let data = '';
        stdin.setEncoding('utf8');
        stdin.on('data', (chunk) => data += chunk);
        stdin.on('end', () => resolve(data));
        stdin.on('error', reject);
    });
}

function checkNPMVersion(npmOutput) {
    const version = npmOutput.trim();

    // Check if using non 6.13.* npm version
    if (!npmOutput.match(/6\.13\..*/g)) {
        throw `
Error: BabylonReactNative Playground development requires npm version 6.13.*.
Your current npm version is ${version}. Run npm install -g npm@6.13 to update your npm version.
            `;
    }
}

getStandardInput().then(checkNPMVersion).catch(console.error);
```
(Source for piping: https://wellingguzman.com/notes/node-pipe-input)

File: Apps/Playground/scripts/postinstall.js [hunk 4]
```diff
@@ -0,0 +1,6 @@
+const shelljs = require('shelljs');
+const version = shelljs.exec('npm --version');
+
+if (!version.trim().match(/6\.\d+\.\d+/g)) {
```
Comment (ryantrem) [inline Apps/Playground/scripts/postinstall.js:4]: Sorry I know there has already been some discussion on this and I meant to comment on it earlier but forgot... you could also consider using the `semver` package, which would make this code a bit more readable. Something like:
```js
const semver = require('semver');
const shelljs = require('shelljs');
const version = semver.clean(shelljs.exec('npm --version'));

if (!semver.lt(version, '7')) {
    throw ...
}
```
Or you might need to do `semver.lt(version, semver.coerce('7'))`, not sure. Anyway up to you, but the `semver` package is all about dealing with versions and version comparisons.

Comment (chrisfromwork) [reply Apps/Playground/scripts/postinstall.js:4]: sounds good. going to go with the regex for now, we can update this in the future as needed!

**Pull request merged**